### PR TITLE
Add convenience script for working in stability branches

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

It's inconvenient to work in stability branches with their toolchains pinned to older versions than are likely to be default in the dev env.  This leads to unintentional `Cargo.lock`, clippy and fmt changes

#### Summary of Changes

Add a convenience script that wraps `cargo`, injecting toolchains as specified in `ci/rust-version.sh`

Alternative naming suggestions welcome :wink:

### CODE REVIEW

File: scripts/curgo.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    shift
+    ;;
+  *)
+    # shellcheck disable=SC2054 # rust_stable is sourced from rust-version.sh
+    toolchain="$rust_stable"
+    ;;
+esac
+
+set -x
+cargo "+${toolchain}" "${@}"
```
Comment (mvines) [inline scripts/curgo.sh:27]:
```suggestion
exec cargo "+${toolchain}" "${@}"
```

**Review by mvines — COMMENTED** [review]
lol that name.  
wdyt about just `./cargo` at the root?

**Comment by t-nelson** [comment]
> lol that name.

:upside_down_face: 

> wdyt about just `./cargo` at the root?

WFM!

**Comment by ryoqun** [comment]
(cool! well, you beat me to it! I always wanted like this.... always buffered in my brain, not flushed ever.. ;))

**Comment by t-nelson** [comment]
Hehe, I finally got tired of `git clean`ing it away!

I'll try to work on a patch set to plumb it through our other scripts so there are no more surprises when switching branches

**Pull request merged**